mh/pat.py

```diff
@@ -2405,6 +2424,19 @@ def notify_layer_departure(self):
             ntc_data = pati.lp2_string(self.session.capcom_id)
             self.server.layer_broadcast(self.session, PatID4.NtcLayerOut,
                                         ntc_data, 0)
+        if self.session.local_info['circle_id'] is not None:
```

Comment:
If `circle_id` can't be zero the following should be used:
```python
if self.session.local_info['circle_id']:
```

Reply:
it can be zero due to python indexing, we add 1 to it before we send it off

Reply:
After checking the code on python, there were no assert in the `circles` code in the database module. So the `session` module (which is caching server data)'s  `circle_id` doesn't hold the server value but our own which is bad. Rather than rewriting all this part I suggest you add a comment:
```python
# TODO: Fix local_info['circle_id'] mistakenly using python indexing
```

Reply:
After addressing @sepalani comments, this `if` block
```py
if self.session.local_info['circle_id'] is not None:
```

Should be moved inside the `if` block
```py
if self.session.layer == 2:
```

Since, it doesn't make sense that after you leave all the others layer, you then leave the circle.

Reply:
They're separated out to decrease the likelihood of errors within the circle code (such as a blank circle id inside a Session) from causing too much collateral damage if they cause this entire function to quit early due to an Exception.